maitake-sync/src/wait_cell.rs

```diff
@@ -327,6 +327,46 @@ impl WaitCell {
         }
     }
 
+    /// Asynchronously poll the [`WaitCell`] until a condition occurs
```

Comment:
Personally, I don't love this wording --- I think it would be more accurate to say that we are polling the _condition_, not the `WaitCell`. This function polls for the condition and uses the waitcell to _wait_ if that condition is not yet met.

Reply:
Also, it might be good for these docs to mention that this method ensures that the future is registered to be woken up by the `WaitCell` *before* checking the condition, so that there isn't a potential race where a wakeup is missed while the condition is being checked, since that's...basically the whole point of having this method :)

It might even be worth comparing to `std::sync::Condvar`, which uses a Mutex to make similar guarantees?